forte/data/ontology/code_generation_objects.py

```diff
@@ -382,6 +382,45 @@ def to_field_value(self):
         return self.name
 
 
+class NdArrayProperty(Property):
```

Comment:
The current implementation is basically a copy of DictProperty, which isn't what we want. What we need to do here is to design an `NDarray` spec language, and then translate that to code. Here, you used `value_type`, what's its purpose? In a `DictProperty`, we use `key_type` and `value_type` because that's what a `Dict` looks like (it has key and value so we ask for key type and value_type).

In an `NDarray`, there shouldn't be values for users to provide, we should only accept types that are available for a numpy array (e.g. int64, float, str etc). If the `value_type` is for the `numpy.ndarray`, then we also do not want the schema to provide it, we should hard-code this internally (perhaps use a variable internally `ndarry_class` for this).